netutils/netinit/netinit.c

```diff
@@ -344,66 +340,239 @@ static void netinit_set_macaddr(void)
 #endif
 
 /****************************************************************************
- * Name: netinit_set_ipaddrs
+ * Name: netinit_set_ipv4addrs
  *
  * Description:
- *   Setup IP addresses.
- *
- *   For 6LoWPAN, the IP address derives from the MAC address.  Setting it
- *   to any user provided value is asking for trouble.
+ *   Setup IPv4 addresses.
  *
  ****************************************************************************/
 
 #if defined(NETINIT_HAVE_NETDEV) && !defined(CONFIG_NET_6LOWPAN) && ! \
-    defined(CONFIG_NET_IEEE802154)
-static void netinit_set_ipaddrs(void)
+    defined(CONFIG_NET_IEEE802154) && defined(CONFIG_NET_IPv4)
+static inline void netinit_set_ipv4addrs(void)
 {
-#ifdef CONFIG_NET_IPv4
   struct in_addr addr;
+#ifdef CONFIG_FSUTILS_IPCFG
+  struct ipv4cfg_s ipv4cfg;
+  int ret;
+#endif
 
-  /* Set up our host address */
+#ifdef CONFIG_FSUTILS_IPCFG
+  /* Attempt to obtain IPv4 address configuration from the IP configuration
+   * file.
+   */
 
-#ifndef CONFIG_NETINIT_DHCPC
-  addr.s_addr = HTONL(CONFIG_NETINIT_IPADDR);
+  ret = ipcfg_read(NET_DEVNAME, (FAR struct ipcfg_s *)&ipv4cfg, AF_INET);
+#ifdef CONFIG_NETUTILS_DHCPC
+  if (ret >= 0 && ipv4cfg.proto != IPv4PROTO_NONE)
 #else
-  addr.s_addr = 0;
+  if (ret >= 0 && IPCFG_HAVE_STATIC(ipv4cfg.proto))
 #endif
-  netlib_set_ipv4addr(NET_DEVNAME, &addr);
+    {
+      /* Check if we are using DHCPC */
 
-  /* Set up the default router address */
+#ifdef CONFIG_NETUTILS_DHCPC
+      if (IPCFG_USE_DHCPC(ipv4cfg.proto))
+        {
+          g_use_dhcpc = true;
+          addr.s_addr = 0;
+        }
+      else
+#endif
+        {
+          /* We are not using DHCPC.  We need an IP address */
 
-  addr.s_addr = HTONL(CONFIG_NETINIT_DRIPADDR);
-  netlib_set_dripv4addr(NET_DEVNAME, &addr);
+#ifdef CONFIG_NETINIT_IPADDR
+          /* Check if we have a static IP address in the configuration file */
 
-  /* Setup the subnet mask */
+          if (ipv4cfg.proto != 0)
```

Comment:
in the DHCP case the `else` limits the scope but would it it make sense to check the static bit here?

Reply:
The test` if (ipv4cfg.proto != 0)` is equivalent but a little stronger.  If the STATIC bit is not set, the addresses should all be zero, but not necessarily.

But I think you are right, better to test the static bit because if the user wants to force a zero address I suppose that should be permitted.

Reply:
Changes per your suggestion.